com.unity.visualeffectgraph/Editor/ShaderGraph/VFXShaderGraphParticleOutput.cs

```diff
@@ -748,6 +746,15 @@ public override void EndCompilation()
                 graphCodes.Clear();
         }
 
+        private static bool NeedsPositionWorldInterpolator(GraphCode graphCode)
+        {
+            return graphCode.requirements.requiresPosition != NeededCoordinateSpace.None
+                    || graphCode.requirements.requiresViewDir != NeededCoordinateSpace.None
+                    || graphCode.requirements.requiresScreenPosition
+                    || graphCode.requirements.requiresNDCPosition
+                    || graphCode.requirements.requiresPixelPosition;
+        }
```

Comment:
I'm not that familiar with SG but why would a world pos interpolator be required in some of those cases? SV_Position seems enough.

Reply:
I mean isnt it exactly what the incriminated PR was fixing?

Reply:
The issue has been introduced precisely at this change: https://github.com/Unity-Technologies/Graphics/commit/d74237b8a3f6e41bfa091ed6bff60014efe32d13#diff-12c1990cad6d2bf349a05118c2befeb419e8bb047aed21d85c30600aac5c554f

The problem is about the condition of VFX_NEEDS_POSWS_INTERPOLATOR define, it used this set of test :
```
graphCode.requirements.requiresPosition != 
NeededCoordinateSpace.None 
|| graphCode.requirements.requiresScreenPosition 
|| graphCode.requirements.requiresViewDir != NeededCoordinateSpace.None;
```

While the condition for generating at call to `VFXGetPositionRWS(i.VFX_VARYING_POSWS);` is now :
```
graphCode.requirements.requiresPosition != NeededCoordinateSpace.None 
|| graphCode.requirements.requiresViewDir != NeededCoordinateSpace.None 
|| graphCode.requirements.requiresScreenPosition 
|| graphCode.requirements.requiresNDCPosition 
|| graphCode.requirements.requiresPixelPosition
```

If `VFX_NEEDS_POSWS_INTERPOLATOR` is defined, then `VFX_VARYING_POSWS` is available.

The computation of old .ScreenPosition still uses a not optimal approach:
```
INSG.ScreenPosition = ComputeScreenPos(VFXTransformPositionWorldToClip(i.VFX_VARYING_POSWS), _ProjectionParams.x);
```

Reply:
Ok but the correct fix would be the same as #5428 ie use SV_Position instead of WorldPosition where relevant. It does not make sense to compute screenspace pos from World pos in fragment when it's available in SV_Position already. It's more expensive and introduce errors. I didnt know it was made like that in SG in the first place.

Reply:
👍 I will extract the code which doesn't use `VFX_VARYING_POSWS` and remove `graphCode.requirements.requiresNDCPosition || graphCode.requirements.requiresPixelPosition` from NeedsPositionWorldInterpolator condition.

Reply:
Fixed at c78ff305c8caf4800881c3a318a6b9e7607fc7f4